samples/core/train_until_good/train_until_good.py

```diff
@@ -0,0 +1,88 @@
+drop_header_op = kfp.components.load_component_from_url('https://raw.githubusercontent.com/kubeflow/pipelines/02c9638287468c849632cf9f7885b51de4c66f86/components/tables/Remove_header/component.yaml')
+calculate_regression_metrics_from_csv_op = kfp.components.load_component_from_url('https://raw.githubusercontent.com/kubeflow/pipelines/616542ac0f789914f4eb53438da713dd3004fba4/components/ml_metrics/Calculate_regression_metrics/from_CSV/component.yaml')
+
+
+# This recursive sub-pipeline trains a model, evaluates it, calculates the metrics and checks them.
+# If the model error is too high, then more training is performed until the model is good.
+@kfp.dsl.graph_component
+def train_until_low_error(starting_model, training_data, true_values):
+    # Training
+    model = xgboost_train_on_csv_op(
+        training_data=training_data,
+        starting_model=starting_model,
+        label_column=0,
+        objective='reg:squarederror',
+        num_iterations=50,
+    ).outputs['model']
+
+    # Predicting
+    predictions = xgboost_predict_on_csv_op(
+        data=training_data,
+        model=model,
+        label_column=0,
+    ).output
+
+    # Calculating the regression metrics    
+    metrics_task = calculate_regression_metrics_from_csv_op(
+        true_values=true_values,
+        predicted_values=predictions,
+    )
+
+    # Checking the metrics
+    with kfp.dsl.Condition(metrics_task.outputs['mean_squared_error'] > 0.01):
+        # Training some more
+        train_until_low_error(
+            starting_model=model,
+            training_data=training_data,
+            true_values=true_values,
+        )
+
+
+# The main pipleine trains the initial model and then gradually trains the model some more until the model evaluation metrics are good enough.
+def train_until_good_pipeline():
+    # Preparing the training data
+    training_data = chicago_taxi_dataset_op(
+        where='trip_start_timestamp >= "2019-01-01" AND trip_start_timestamp < "2019-02-01"',
+        select='tips,trip_seconds,trip_miles,pickup_community_area,dropoff_community_area,fare,tolls,extras,trip_total',
+        limit=10000,
+    ).output
+
+    # Preparing the true values
+    true_values_table = pandas_transform_csv_op(
+        table=training_data,
+        transform_code='''df = df[["tips"]]''',
+    ).output
+    
+    true_values = drop_header_op(true_values_table).output
+
+    # Initial model training
+    first_model = xgboost_train_on_csv_op(
+        training_data=training_data,
+        label_column=0,
+        objective='reg:squarederror',
+        num_iterations=100,
+    ).outputs['model']
+
+    # Recursively training until the error becomes low
+    train_until_low_error(
+        starting_model=first_model,
+        training_data=training_data,
+        true_values=true_values,
+    )
+
+
+if __name__ == '__main__':
+    kfp_endpoint=None
+    kfp.Client(host=kfp_endpoint).create_run_from_pipeline_func(train_until_good_pipeline, arguments={})
```

Comment:
Currently other core samples output a compiled pipeline yaml spec, which is more compatible with the current test infra. Shall we also make this sample output a yaml instead of kicking off an execution? So that it's easier to add sample test later on.

Reply:
Hmm. I'll think of the best way to do this. I wonder whether we can submit those kind of samples directly.

Reply:
I think I'll solve this issue when I'll be adding this sample to sample tests.
I'm trying to keep the samples for users first and tests second.

Reply:
Sounds good.